### PR TITLE
Fix Analytic orderBy and partitionBy to follow the SQL documentation

### PR DESCRIPTION
**Pull request by wolfcomp**
This is a little annoyance I found while writing query calls using `rowNumber` as the typescript types shows you are allowed to do this but if you try to use an array of objects the SQL request fail with `column ?? does not exist` where ?? is the array of objects columnized.

### CODE REVIEW

**Comment by kibertoad** [comment]
Thank you for your contribution! Could you please add integration test for this?

**Comment by kibertoad** [comment]
@wolfcomp Can I help you with writing a test?

**Comment by wolfcomp** [comment]
Currently writing the tests. Im also adding it into the `partition by` statement at the same time, as they use the exact same operations in the SQL documentation.

**Comment by wolfcomp** [comment]
@kibertoad if you could start the workflows believe it should work as i cant see it in my terminal window due to me not being able to setup sql server connections on the network im on currently

**Comment by wolfcomp** [comment]
Im unsure to why its saying that the formatter is undefined for columnize as the ones that fails seems to have the formatter set.

**Comment by kibertoad** [comment]
formater -> formatter

**Comment by wolfcomp** [comment]
🤦

**Comment by wolfcomp** [comment]
Now it should work surprised that I haven't had any errors my self.

**Comment by wolfcomp** [comment]
Actually there seems to still be an error as in the project that I use this I did function and not object assign on the parameter.

**Comment by wolfcomp** [comment]
@kibertoad now it should all be fixed from what I can see in my terminal window

File: lib/query/analytic.js [hunk 1]
```diff
@@ -18,7 +19,7 @@ class Analytic {
     this.grouping = 'columns';
   }
 
-  partitionBy(column) {
+  partitionBy(column, direction) {
```
Comment (kibertoad) [inline lib/query/analytic.js:22]: This needs to be documented. Could you create a PR for https://github.com/knex/documentation?

File: lib/query/querycompiler.js [hunk 2]
```diff
@@ -1035,13 +1035,23 @@ class QueryCompiler {
         sql += 'partition by ';
         sql +=
           map(stmt.partitions, function (partition) {
-            return self.formatter.columnize(partition);
+            if (isString(partition)) return self.formatter.columnize(partition);
```
Comment (kibertoad) [inline lib/query/querycompiler.js:1038]: Please add missing {}

File: lib/query/querycompiler.js [hunk 3]
```diff
@@ -1035,13 +1035,23 @@ class QueryCompiler {
...
+                self.formatter.columnize(partition.column) +
+                (partition.order ? ' ' + partition.order : '')
+              );
           }).join(', ') + ' ';
       }
 
       sql += 'order by ';
       sql += map(stmt.order, function (order) {
-        return self.formatter.columnize(order);
+        if (isString(order)) return self.formatter.columnize(order);
```
Comment (kibertoad) [inline lib/query/querycompiler.js:1049]: Please add missing {}

File: test/unit/query/builder.js [hunk 4]
```diff
@@ -7854,6 +7854,29 @@ describe('QueryBuilder', () => {
     );
   });
 
+  it('row_number with object', function () {
```
Comment (kibertoad) [inline test/unit/query/builder.js:7857]: can you also add an integration test?

Comment (wolfcomp) [reply test/unit/query/builder.js:7857]: I can try to look into it, haven't really written integration tests before, probably will just copy one of the other ones and get it to fit the function

Comment (wolfcomp) [reply test/unit/query/builder.js:7857]: @kibertoad after looking through the `test/integration` folder, I can't find anything previously for `rowNumber`, `denseRank`, `rank` in any of the files, so I'm unsure of how I should do it in that case.

File: lib/query/analytic.js (hunk 1 above)
Comment (wolfcomp) [reply lib/query/analytic.js:21]: https://github.com/knex/documentation/pull/333

File: test/unit/query/builder.js (hunk 4 above)
Comment (kibertoad) [reply test/unit/query/builder.js:7857]: You are right, we only have unit tests for analytic functions now

**Comment by kibertoad** [comment]
@wolfcomp Are TS types correct now? Could you add a tsd test for this case?

**Comment by wolfcomp** [comment]
I can check if the ts files are correct if nor correct them fast

**Comment by wolfcomp** [comment]
There seems to be no tsd tests for analytic functions at all from what I can find.

**Comment by wolfcomp** [comment]
@kibertoad this is what the best I can do from what I can find in the files when it comes to already existing code for testing and documenting [knex/documentation#333](https://github.com/knex/documentation/pull/333) the added and corrected functionality

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Released in 0.95.11

**Pull request merged**